### ISSUE TITLE
`Kokkos::RangePolicy` and `Kokkos_AnalyzePolicy` : compilation should fail if unmatched trait provided

### ISSUE THREAD

**Issue by romintomasetti** [kokkos/kokkos#5223]
**Describe the bug**

The following code snippet should fail to compile
```c++
using rp = Kokkos::RangePolicy<Kokkos::HostSpace>
```
because `Kokkos::HostSpace` doesn't match any of the `execution_policy_trait_specifications`, and we though it should fail on https://github.com/kokkos/kokkos/blob/d19aab9981a2c447e832a7b4eb7b16992328fb14/core/src/impl/Kokkos_AnalyzePolicy.hpp#L173.

It didn't fail, and `rp` is therefore equivalent to `Kokkos::RangePolicy<>`.

**Please also include the following items to support reproducing the bug**
1. `nvcc 11.6` and `gcc 9.4.0`
2. Kokkos release or commit used [ea2a5f8bab8478ac2e450548c809e522ac665fab](https://github.com/kokkos/kokkos/commit/ea2a5f8bab8478ac2e450548c809e522ac665fab)
3. `Ubuntu 22`

**Comment by dalg24** [comment]
It gets interpreted as a tag
```C++
  using Policy = Kokkos::RangePolicy<Kokkos::HostSpace>;
  static_assert(std::is_same<Policy::work_tag, Kokkos::HostSpace>::value, "");
```
(No judgement whether it is right or not here, just saying how it is interpreted by Kokkos)

**Comment by dalg24** [comment]
(Sorry for the digression) This reminds me that I contemplated adding a requirement on the tag that `std::is_empty<WorTag>::value` is `true` to avoid surprising users but was hesitant to proceed because it would potentially break user code.
I believe the current requirements, even though not spelled out in the doc nor by a well placed static assertion, are something like being default constructible on host/device, and of course (sigh) that may depend on the backend implementation.

**Comment by dalg24** [comment]
This is fixed in `develop` and will be in release 3.7
